prime-router/docs/standard-operating-procedures/adding-conformance-profile.md

```diff
@@ -0,0 +1,86 @@
+2. The structure used to parse a given HL7 message is determined in `HL7Reader.getMessageModelClasses()`. Review that
+   function to ensure the combination of `typeID` and `profileID` will return the appropriate structure.
+
+## Mapping
+
+When looking to add a new mapping suite the first step to review existing mapping suites with an eye for reusability.
+Mappings are structured with base mappings separate from mappings that deviate from the base specification.
+
+You can see the separation in the mapping for v251-elr. That conformance profile is closely aligned with the profile for
+v251. That presents the opportunity for reuse.
+
+### HL7 to FHIR
+
+1. Review existing data type mappings located in `prime-router/metadata/HL7/datatypes/` for reuse
+2. Add folder to `prime-router/metadata/HL7/` with a name indicating the new conformance profile
+3. Copy folder `fhir` from `prime-router/metadata/HL7/v251-elr` and add to your new mapping folder.
+4. Create folder `hl7` in your new mapping folder with the following subfolders:
+    - "message" - **required** - within that, add a yml file named with the HL7 message type (ex ORU_R01, ORM_O01,
+      ADT_A01, etc)
+    - "resource" - **required** - within that, add an empty file `Common.yml`. This file is required as an overwrite of
+      the library.
+    - "segments" - **required** - within that, add folders for segments identified in the conformance profile
+    - "codesystem" - **required** - two files are needed in this folder: `CodingSystemMapping.yml`
+      and `ExtensionUrlMapping.yml`. Review other mapping suites and copy those files to use as a starting place then
+      update as needed.
+    - "datatypes" - **optional** - this folder will hold any new datatypes or datatypes that deviate from the base
+      datatypes
+5. Starting with message type file in your `/message` directory build the mappings.
+   The [Templating Guide](https://github.com/LinuxForHealth/hl7v2-fhir-converter/blob/master/TEMPLATING.md) can be
+   helpful.
+
+### FHIR to HL7
+
+1. Review existing datatype mappings located in `prime-router/src/main/resources/metadata/hl7_mapping`
+2. Add folder to `prime-router/src/main/resources/metadata/hl7_mapping` with a name indicating the new conformance
+   profile
+3. Add YML file for the HL7 message type (ex ORU_R01, ORM_O01, ADT_A01, etc)
+    - This file will be the starting point for mapping
+4. Create a folder named "base". This will hold folders and sub-folders for the HL7 message groups. See
+   the `prime-router/src/main/resources/metadata/hl7_mapping/v251-elr/` for an example
+5. Create a folder "resources". This will hold yaml files from mapping directly from FHIR resources
+6. Create a folder "datatypes". This folder will hold any new datatypes or datatypes that deviate from the base
+   datatypes
+
+#### Resources
+
+- The FHIR to HL7 mappings extensively use [FHIR Path](https://hl7.org/fhirpath/N1/) to access values in FHIR
```

Comment:
I don't know how out of date this doc has been made by the recent file shuffle, but might be good to link here? https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/design/design/mapping-schemas.md

Reply:
another good call, also added